app/src/main/kotlin/no/nav/k9punsj/domenetjenester/mappers/MapOmsMATilK9Format.kt

```diff
@@ -0,0 +1,119 @@
+import no.nav.k9.søknad.ytelse.omsorgspenger.utvidetrett.v1.OmsorgspengerKroniskSyktBarn
+import no.nav.k9.søknad.ytelse.omsorgspenger.utvidetrett.v1.OmsorgspengerMidlertidigAlene
+import no.nav.k9punsj.rest.web.JournalpostId
+import no.nav.k9punsj.rest.web.dto.OmsorgspengerMidlertidigAleneSøknadDto
+import no.nav.k9punsj.rest.web.dto.PeriodeDto
+import org.slf4j.LoggerFactory
+import java.time.ZoneId
+import java.time.ZonedDateTime
+
+internal class MapOmsMATilK9Format(
+    søknadId: String,
+    journalpostIder: Set<String>,
+    dto: OmsorgspengerMidlertidigAleneSøknadDto,
+) {
+    private val søknad = Søknad()
+    private val omsorgspengerMidlertidigAlene = OmsorgspengerMidlertidigAlene()
+    private val feil = mutableListOf<Feil>()
+
+    init {
+        kotlin.runCatching {
+            søknadId.leggTilSøknadId()
+            Versjon.leggTilVersjon()
+            dto.leggTilMottattDato()
+            dto.soekerId?.leggTilSøker()
+            dto.barn.leggTilBarn()
+            dto.annenForelder?.leggTilAnnenForelder()
+            dto.leggTilJournalposter(journalpostIder = journalpostIder)
+
+            // Fullfører søknad & validerer
+            søknad.medYtelse(omsorgspengerMidlertidigAlene)
+            feil.addAll(Validator.valider(søknad.getYtelse())) // TODO: 20/01/2022 Validerer ingenting...
+        }.onFailure { throwable ->
+            logger.error("Uventet mappingfeil", throwable)
+            feil.add(Feil("søknad", "uventetMappingfeil", throwable.message ?: "Uventet mappingfeil"))
+        }
+    }
+
+    internal fun søknad() = søknad
+    internal fun feil() = feil.toList()
+    internal fun søknadOgFeil() = søknad() to feil()
+
+    private fun String.leggTilSøknadId() {
+        if (erSatt()) {
+            søknad.medSøknadId(this)
+        }
+    }
+
+    private fun String.leggTilVersjon() {
+        søknad.medVersjon(this)
+    }
+
+    private fun OmsorgspengerMidlertidigAleneSøknadDto.leggTilMottattDato() {
+        if (mottattDato != null && klokkeslett != null) {
+            søknad.medMottattDato(ZonedDateTime.of(mottattDato, klokkeslett, Oslo))
+        }
+    }
+
+    private fun OmsorgspengerMidlertidigAleneSøknadDto.AnnenForelder.leggTilAnnenForelder() {
+        val annenForelder = AnnenForelder()
+        annenForelder.medPeriode(this.periode?.somK9Periode())
+        val situasjonType = this.situasjonType
+        if (situasjonType != null) {
+            annenForelder.medSituasjon(AnnenForelder.SituasjonType.valueOf(situasjonType), this.situasjonBeskrivelse)
+        }
+        annenForelder.medNorskIdentitetsnummer(NorskIdentitetsnummer.of(this.norskIdent))
+        omsorgspengerMidlertidigAlene.medAnnenForelder(annenForelder)
+    }
+
+    private fun List<OmsorgspengerMidlertidigAleneSøknadDto.BarnDto>.leggTilBarn() {
+        val barnListe = this.map {
+            val barn = Barn()
+            barn.medFødselsdato(it.foedselsdato)
```

Comment:
Burde det ikke sjekkes for hva som er satt her før man setter fødselsdato og norskIdent?
Dette kan potensielt føre til en utvetydig valideringsfeil fra k9Format på barnet dersom både fødselsdato og norskIdent er satt.

Reply:
Er ikke noe problemet at både fnr og fødselsdato er satt. så lenge en av de er satt :)

Reply:
private boolean isOk() {
        return this.norskIdentitetsnummer != null && this.norskIdentitetsnummer.getVerdi() != null || this.fødselsdato != null;
    }

Reply:
Ok. Har vært borti at k9Format har gitt valideringsfeil dersom begge er satt :)